impl/sql/mutationstorage/queue.go

```diff
@@ -81,37 +82,43 @@ func (r *Reciever) run(ctx context.Context, last time.Time) {
 	r.finished.Add(1)
 	defer r.finished.Done()
 
+	var count int32
 	more := make(chan bool, 1)
 	if time.Since(last) > (r.opts.MaxPeriod - r.opts.Period) {
-		more <- true // We will be over due for an epoch soon.
+		count = r.sendBatch(ctx, true) // We will be over due for an epoch soon.
 	}
 
 	for {
 		select {
 		case <-more:
+			count = r.sendBatch(ctx, false)
 		case <-r.ticker.C:
+			count = r.sendBatch(ctx, false)
 		case <-r.maxTicker.C:
```

Comment:
So, for testing, I wonder about defining a helper func:
```
func maxTickerChan(r *Receiver) chan(interface{}) {
  return r.maxTicker.C
}
```
which somehow you replace in the test code:

```
maxTickerChan = func(_ *Receiver) chan(interface{}) {
  return myTestChan
}
```

You can then signal `myTestChan` in the test code to jog things along.
This doesn't quite feel right but you get the idea.